### PR TITLE
Export vshard config API

### LINKED ISSUE #1761: Export vshard config in Lua API
Hello

There are cases when cartridge tests touches vshard corner cases and it will be better to move such cases to vshard repo.
So, I propose to export vshard config from cartridge lua api to make it easier.

### PR DESCRIPTION
**Pull request by panticonur**
Create public API to get vshard config.

I didn't forget about

- [x] Tests
- [x] Changelog

Close #1761

### CODE REVIEW

File: cartridge/lua-api/vshard.lua [hunk 1]
```diff
@@ -3,6 +3,33 @@
 -- @module cartridge.lua-api.vshard
 
 local rpc = require('cartridge.rpc')
+local errors = require('errors')
+local confapplier = require('cartridge.confapplier')
+local vshard_utils = require('cartridge.vshard-utils')
+
+local e_vshard = errors.new_class('vshard api error')
```
Comment (yngvar-antonsson) [inline cartridge/lua-api/vshard.lua:10]: https://github.com/tarantool/errors#naming-conventions

File: test/integration/vshard_api_test.lua [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                servers = {{
+                    alias = 'A1',
+                    instance_uuid = helpers.uuid('a', 'a', 1),
+                    advertise_port = 13301,
+                    http_port = 8081,
+                }}
+            }, {
+                uuid = helpers.uuid('b'),
+                roles = {'myrole'},
+                servers = {
```
Comment (yngvar-antonsson) [inline test/integration/vshard_api_test.lua:25]: please compact test:

```lua
{
                 roles = {'myrole'},
                 servers = 2
}
```

File: test/integration/vshard_api_test.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+local t = require('luatest')
+local g = t.group()
+
+local helpers = require('test.helper')
+
+g.before_all = function()
+    g.cluster = helpers.Cluster:new({
+        datadir = fio.tempdir(),
+        server_command = helpers.entrypoint('srv_basic'),
+        cookie = require('digest').urandom(6):hex(),
```
Comment (yngvar-antonsson) [inline test/integration/vshard_api_test.lua:11]: helpers.random_cookie()

File: test/integration/vshard_api_test.lua [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    })
+    g.cluster:start()
+end
+
+g.after_all = function()
+    g.cluster:stop()
+    fio.rmtree(g.cluster.datadir)
+end
+
+function g.test_get_cfg()
```
Comment (yngvar-antonsson) [inline test/integration/vshard_api_test.lua:49]: please add a multisharding test. (see example in tests with name multisharding*)

**Review by yngvar-antonsson — CHANGES_REQUESTED** [review]

**Review by yngvar-antonsson — APPROVED** [review]

File: cartridge/lua-api/vshard.lua [hunk 5]
```diff
@@ -3,6 +3,36 @@
...
+-- @treturn[2] table Error description
+local function get_config()
+    local result = {}
+    local conf = confapplier.get_readonly()
+    if conf == nil then
+        return nil, VshardApiError:new('not bootstrapped')
+    end
+    local vshard_groups
+    if conf.vshard_groups == nil then
+        vshard_groups = {default = conf.vshard}
```
[the remark this answers is not in the document]
Comment (filonenko-mikhail) [reply cartridge/lua-api/vshard.lua:25]: No

[the remark this answers is not in the document]
Comment (olegrok) [reply cartridge/lua-api/vshard.lua:25]: But there is `conf = confapplier.*get_readonly()*` above

[the remark this answers is not in the document]
Comment (filonenko-mikhail) [reply cartridge/lua-api/vshard.lua:25]: Conf is not used below as destination. But you are right — code is not clean.
